ProcessLib/UncoupledProcessesTimeLoop.cpp

```diff
@@ -492,41 +612,111 @@ bool UncoupledProcessesTimeLoop::loop()
         INFO("=== timestep #%u (t=%gs, dt=%gs) ==============================",
              timestep, t, delta_t);
 
-        // TODO use process name
-        unsigned pcs_idx = 0;
-        for (auto& spd : _per_process_data)
+        bool coupling_iteration_converged = true;
+        // Coupling iteration
+        for (unsigned i = 0; i < _global_coupling_max_iterations; i++)
```

Comment:
The content of the for loop is now much longer and deeply nested; I'd recommend to move it or parts of it to some functions.